In [11]:
import pandas as pd
data1 = pd.read_excel('mabse.xlsx')
df1 = pd.DataFrame(data1)
df1.sort_values(by = ['mgtkas_date'])
df1 = df1.loc[df1['mgtkas_date']  >= '2012-01-01']
df1.shape

(6435, 10)

In [12]:
data2 = pd.read_excel('bse500companies.xlsx')
df2 = pd.DataFrame(data2)
df2.head()

Scrip Code                          COMPANY  \
0    523395.0                   3M India Ltd.    
1    512161.0  8K Miles Software Services Ltd.   
2    524208.0           Aarti Industries Ltd.    
3    523204.0               Aban Offshore Ltd.   
4    500002.0                   ABB India Ltd.   

                              CAPS  
0                   3M INDIA LTD.   
1  8K MILES SOFTWARE SERVICES LTD.  
2           AARTI INDUSTRIES LTD.   
3               ABAN OFFSHORE LTD.  
4                   ABB INDIA LTD.

In [13]:
list1 = df2['CAPS'].tolist()
df3 = df1[df1['acquirer'].isin(list1)]
list2 = df3['acquirer'].tolist()

In [14]:
df6 = df2[df2['CAPS'].isin(list2)]
df6.head()

Scrip Code                            COMPANY  \
1     512161.0    8K Miles Software Services Ltd.   
3     523204.0                 Aban Offshore Ltd.   
8     533096.0                   Adani Power Ltd.   
9     539254.0            Adani Transmission Ltd.   
11    540025.0  Advanced Enzyme Technologies Ltd.   

                                 CAPS  
1     8K MILES SOFTWARE SERVICES LTD.  
3                  ABAN OFFSHORE LTD.  
8                    ADANI POWER LTD.  
9             ADANI TRANSMISSION LTD.  
11  ADVANCED ENZYME TECHNOLOGIES LTD.

In [15]:
df3.head()

co_code                  company_name      mr_info_full_name  merg_num  \
11117   371246           TV18 BROADCAST LTD.  Acquisition of shares         1   
11127   178651          PRAJ INDUSTRIES LTD.  Acquisition of shares         1   
11165    27747  RELIANCE INFRASTRUCTURE LTD.          Sale of asset         1   
11166    27747  RELIANCE INFRASTRUCTURE LTD.                 Merger         1   
11167    27747  RELIANCE INFRASTRUCTURE LTD.                 Merger         1   

      mgtkas_date                                    entity_name_mst  \
11117  2012-01-03              EQUATOR TRADING ENTERPRISES PVT. LTD.   
11127  2012-01-06                         PRAJ HIPURITY SYSTEMS LTD.   
11165  2012-01-17  RELIANCE INFRASTRUCTURE ENGINEERS PVT. LTD. [M...   
11166  2012-01-17                      RELIANCE ENERGY LTD. [MERGED]   
11167  2012-01-17           RELIANCE ENERGY GENERATION LTD. [MERGED]   

                           acquirer co_advisor merg_acq_advisor  \
11117           TV18 BROADCAST LTD.        NaN              NaN   
11127          PRAJ INDUSTRIES LTD.        NaN              NaN   
11165  RELIANCE INFRASTRUCTURE LTD.        NaN              NaN   
11166  RELIANCE INFRASTRUCTURE LTD.        NaN              NaN   
11167  RELIANCE INFRASTRUCTURE LTD.        NaN              NaN   

      merg_acq_mer_banker  
11117                 NaN  
11127                 NaN  
11165                 NaN  
11166                 NaN  
11167                 NaN

In [16]:
df3 = df3.loc[df3['mr_info_full_name'] == 'Acquisition of shares']
df3.shape


(480, 10)

In [17]:
from datetime import datetime, timedelta
d = {'DATE OF ACQUISTION':[],'DATE one month AGO':[],'DATE ON YEAR ago':[]}
df4 = pd.DataFrame(data = d)
for index, row in df3.iterrows():
    currentdatetime = row['mgtkas_date']
    N = 365
    date_N_days_ago = currentdatetime - timedelta(days= 10)
    date_N_days_later = date_N_days_ago - timedelta(days=N)
    df4 = df4.append({'DATE OF ACQUISTION':currentdatetime,'DATE one month AGO':date_N_days_ago,'DATE ON YEAR ago':date_N_days_later}, ignore_index=True)


480 ACQUISTION CASES WE EXPECT 480 CSV FILES TO BE OUTPUT 

In [19]:
df4.shape

(480, 3)

In [20]:
df5 = pd.merge(df3, df4, how='inner',  left_on='mgtkas_date', right_on = 'DATE OF ACQUISTION')
df5.shape

(1120, 13)

In [22]:
df5

co_code                       company_name      mr_info_full_name  \
0      371246                TV18 BROADCAST LTD.  Acquisition of shares   
1      178651               PRAJ INDUSTRIES LTD.  Acquisition of shares   
2       83013      GODREJ CONSUMER PRODUCTS LTD.  Acquisition of shares   
3       83013      GODREJ CONSUMER PRODUCTS LTD.  Acquisition of shares   
4       83013      GODREJ CONSUMER PRODUCTS LTD.  Acquisition of shares   
5       83013      GODREJ CONSUMER PRODUCTS LTD.  Acquisition of shares   
6      146460                      MINDTREE LTD.  Acquisition of shares   
7      146460                      MINDTREE LTD.  Acquisition of shares   
8      146460                      MINDTREE LTD.  Acquisition of shares   
9      146460                      MINDTREE LTD.  Acquisition of shares   
10     175018                 PHOENIX MILLS LTD.  Acquisition of shares   
11     175018                 PHOENIX MILLS LTD.  Acquisition of shares   
12     175018                 PHOENIX MILLS LTD.  Acquisition of shares   
13     175018                 PHOENIX MILLS LTD.  Acquisition of shares   
14     389273        COFFEE DAY ENTERPRISES LTD.  Acquisition of shares   
15     389273        COFFEE DAY ENTERPRISES LTD.  Acquisition of shares   
16     389273        COFFEE DAY ENTERPRISES LTD.  Acquisition of shares   
17     389273        COFFEE DAY ENTERPRISES LTD.  Acquisition of shares   
18     159120           PIRAMAL ENTERPRISES LTD.  Acquisition of shares   
19      14897                     ANANT RAJ LTD.  Acquisition of shares   
20      14897                     ANANT RAJ LTD.  Acquisition of shares   
21     196588              RELIANCE CAPITAL LTD.  Acquisition of shares   
22     196588              RELIANCE CAPITAL LTD.  Acquisition of shares   
23     251642               TITAGARH WAGONS LTD.  Acquisition of shares   
24     196588              RELIANCE CAPITAL LTD.  Acquisition of shares   
25     196588              RELIANCE CAPITAL LTD.  Acquisition of shares   
26     369944       RELIANCE COMMUNICATIONS LTD.  Acquisition of shares   
27     369944       RELIANCE COMMUNICATIONS LTD.  Acquisition of shares   
28     250369                       THERMAX LTD.  Acquisition of shares   
29     248186         TATA GLOBAL BEVERAGES LTD.  Acquisition of shares   
...       ...                                ...                    ...   
1090   256066                     AXIS BANK LTD.  Acquisition of shares   
1091   256066                     AXIS BANK LTD.  Acquisition of shares   
1092   256066                     AXIS BANK LTD.  Acquisition of shares   
1093   100632                       INFOSYS LTD.  Acquisition of shares   
1094   502255            ADANI TRANSMISSION LTD.  Acquisition of shares   
1095   502255            ADANI TRANSMISSION LTD.  Acquisition of shares   
1096   502255            ADANI TRANSMISSION LTD.  Acquisition of shares   
1097   502255            ADANI TRANSMISSION LTD.  Acquisition of shares   
1098   331940              DR. LAL PATHLABS LTD.  Acquisition of shares   
1099   288148               MUTHOOT FINANCE LTD.  Acquisition of shares   
1100    24685              AUROBINDO PHARMA LTD.  Acquisition of shares   
1101    33750                  BHARAT FORGE LTD.  Acquisition of shares   
1102   502255            ADANI TRANSMISSION LTD.  Acquisition of shares   
1103   261187            V-GUARD INDUSTRIES LTD.  Acquisition of shares   
1104   261187            V-GUARD INDUSTRIES LTD.  Acquisition of shares   
1105   389273        COFFEE DAY ENTERPRISES LTD.  Acquisition of shares   
1106   389273        COFFEE DAY ENTERPRISES LTD.  Acquisition of shares   
1107   150059         BOMBAY RAYON FASHIONS LTD.  Acquisition of shares   
1108   150059         BOMBAY RAYON FASHIONS LTD.  Acquisition of shares   
1109   236328                STATE BANK OF INDIA  Acquisition of shares   
1110   236328                STATE BANK OF INDIA  Acquisition of shares   
1111   146439              MIND

In [23]:
writer = pd.ExcelWriter('result.xlsx')
df5.to_excel(writer,'Sheet1')
writer.save()

In [24]:
data4 = pd.read_excel('result.xlsx')
df6 = pd.DataFrame(data4)
df6.shape

(480, 10)

In [25]:
df6 = df6.sort_values(['mgtkas_date'])

making filename column